lib/server/monalisa-synchronization/MonALISAClient.js

```diff
@@ -157,6 +181,104 @@ class MonALISAClient {
+        const specificURL = new URL(this.simulationPassesUrl);
+        specificURL.searchParams.set('res_path', 'json');
+        const response = await fetch(specificURL, this.httpOptions);
+        return response.json();
+    }
+
+    /**
+     * Extract simulation passes from MonALISA response payload
+     * @param {object<string, Partial<MonALISASimulationPass>>} payload raw data acquired from MonALISA:
+     * mapping of a simulation pass name to its parameters
+     * @return {{properties: SimulationPass, associations: SimulationPassAssociations}[]} adapted data
+     * @private
+     */
+    _extractSimulationPassesFromPayload(payload) {
+        return Object.entries(payload).map(([simulationPassName, simulationPassParameters]) => ({
+            ...simulationPassParameters,
+            name: simulationPassName,
+        }))
+            .filter(({ anchor_production, anchor_pass }) => anchor_production.trim() && anchor_pass.trim())
+            .map((simulationPass) => this._monALISASimulationPassToEntity(simulationPass))
+            .filter(({ associations: { lhcPeriods, dataPassesSuffixes } }) => lhcPeriods.length > 0 && dataPassesSuffixes.length > 0);
+    }
+
+    /**
+     * Extract jira id from simulation pass description
+     * @param {string} description simulation pass description
+     * @return {{extractedJiraId: string, extractedDescription: string}} jira id and lighten description
+     */
+    _ectractJiraIdFromDescription(description) {
+        const tokens = description?.trim(0).split(',');
+        if (!tokens) {
+            return null;
+        }
+        const presumedJiraId = tokens[tokens.length - 1]?.trim();
+        if (/[A-Z]+[A-Z0-9]+-\d+/.test(presumedJiraId)) {
+            return {
+                extractedDescription: description.trim().replace(new RegExp(`, *${presumedJiraId}$`), ''),
+                extractedJiraId: presumedJiraId,
+            };
+        } else {
+            return { extractedDescription: description?.trim(), extractedJiraId: null };
+        }
+    }
+
+    /**
+     * Adapt simulation pass
+     * @param {MonALISASimulationPass} simulationPass simulation pass
+     * @return {{properties: SimulationPass, associations: SimulationPassAssociations}}
+     * Convert MonALISA simulation pass to simulation pass entity with its association data
+     * @private
+     */
+    _monALISASimulationPassToEntity(simulationPass) {
+        const { extractedDescription, extractedJiraId } = this._ectractJiraIdFromDescription(simulationPass.generator);
+        return {
+            properties: {
+                name: simulationPass.name,
+                jiraId: extractedJiraId || simulationPass.jiraID.trim() || null,
+                description: extractedDescription || null,
+                pwg: simulationPass.PWG?.trim() || null,
+                requestedEventsCount: Number(simulationPass.requested_events) || null, // 0 is not valid value so it's turned to null
```

Comment:
I see really few reasons to explicitly convert 0 to null. If this happens, why not leaving it to 0? It will not break the code, and it may be a good thing that people catch it

Reply:
well, ok, coalesce to null  removed